registry/registry.go

```diff
@@ -324,15 +323,20 @@ func (registry *Registry) ListenAndServe() error {
 	select {
 	case err := <-serveErr:
 		return err
-	case <-quit:
-		dcontext.GetLogger(registry.app).Info("stopping server gracefully. Draining connections for ", config.HTTP.DrainTimeout)
-		// shutdown the server with a grace period of configured timeout
-		c, cancel := context.WithTimeout(context.Background(), config.HTTP.DrainTimeout)
-		defer cancel()
-		return registry.server.Shutdown(c)
+	case <-registry.quit:
+		return registry.Shutdown()
 	}
 }
 
+// Shutdown gracefully shuts down the registry's HTTP server.
+func (registry *Registry) Shutdown() error {
+	dcontext.GetLogger(registry.app).Info("stopping server gracefully. Draining connections for ", registry.config.HTTP.DrainTimeout)
+	// shutdown the server with a grace period of configured timeout
+	c, cancel := context.WithTimeout(context.Background(), registry.config.HTTP.DrainTimeout)
```

Comment:
As this is new functionality; would it make sense to pass a context to the function; also to make it closer to the `server.Shutdown()` ?

Reply:
I modified to function to accept a context instead of setting its own. Please take a look.